overwatch-derive/src/utils.rs

```diff
@@ -32,3 +32,34 @@ pub fn extract_type_from(ty: &Type) -> Type {
         _ => abort_call_site!("Expected single type argument, found {}", stringify_type),
     }
 }
+
+pub fn field_name_to_type_name(name: &str) -> String {
+    let mut result = String::new();
+    let mut capitalize_next = true;
+
+    for symbol in name.chars() {
+        if symbol == '_' {
+            // Skip the underscore, and mark the next letter for capitalization
+            capitalize_next = true;
+        } else if capitalize_next {
+            // Capitalize the current character
+            result.push(
+                symbol
+                    .to_uppercase()
+                    .next()
+                    .expect("Name cannot end with an underscore."),
+            );
+            capitalize_next = false;
+        } else {
+            // Append the character as-is (lowercase)
+            result.push(
+                symbol
+                    .to_lowercase()
+                    .next()
+                    .expect("Name cannot end with an underscore."),
+            );
+        }
+    }
+
+    result
+}
```

Comment:
Can probably be replace by: https://crates.io/crates/convert_case

Reply:
Nice! https://github.com/logos-co/Overwatch/pull/58/commits/f2d2509c434911a60a65742a146f6a1024267a53